android/ObjectPermissionAdvanced/app/src/main/java/io/realm/chat/util/PermissionHelper.java

```diff
@@ -0,0 +1,140 @@
+package io.realm.chat.util;
+
+import java.util.List;
+
+import io.realm.Realm;
+import io.realm.RealmResults;
+import io.realm.chat.model.PrivateChatRoom;
+import io.realm.chat.ui.GrantedPermission;
+import io.realm.sync.permissions.ClassPermissions;
+import io.realm.sync.permissions.Permission;
+import io.realm.sync.permissions.PermissionUser;
+import io.realm.sync.permissions.RealmPermissions;
+import io.realm.sync.permissions.Role;
+
+/**
+ * A helper class defining permission related actions.
+ */
+public class PermissionHelper {
+
+    /**
+     * Configure the permissions on the Realm and each model class.
+     * This will only succeed the first time that this code is executed. Subsequent attempts
+     * will silently fail due to `canSetPermissions` having already been removed.
+     *
+     * @param postInitialization block to run after the Role has been added or found, usually a
+     *                           navigation to the next screen.
+     */
+    public static void initializePermissions(Runnable postInitialization) {
+        Realm realm = Realm.getDefaultInstance();
+        RealmResults<RealmPermissions> realmPermissions = realm.where(RealmPermissions.class).findAllAsync();
+        Permission realmPermission = realmPermissions.first().getPermissions().first();
```

Comment:
Something isn't right where. You run an async query and then proceed to directly use the unloaded result.

Reply:
nice catch!